src/dumpContent.ts

```diff
@@ -0,0 +1,161 @@
+    'Please provide CONTENT_MASTER_SCRIVITO_TENANT and credentials:',
+    'CONTENT_MASTER_API_CLIENT_ID, CONTENT_MASTER_API_CLIENT_SECRET.',
+  )
+  process.exitCode = -1
+}
+
+function clearDump() {
+  fs.rmSync(DUMP_PATH, { force: true, recursive: true })
+  fs.mkdirSync(OBJS_PATH, { recursive: true })
+  fs.mkdirSync(BINARIES_PATH, { recursive: true })
+}
+
+function fileStats() {
+  const objs = fs.readdirSync(OBJS_PATH)
+  const binaries = fs.readdirSync(BINARIES_PATH)
+  return `${objs.length} objs and ${binaries.length} binaries`
+}
+
+async function dumpContent() {
+  let continuation: string | undefined
+
+  do {
+    const data: SearchData = await fetchJson<SearchData>(
+      `workspaces/published/objs/search`,
+      {
+        data: {
+          continuation,
+          include_objs: true,
+          options: { site_aware: true },
+          size: 10,
+        },
+        method: 'put',
+      },
+    )
+
+    for (const objData of data.objs) await dumpObjAndBinaries(objData)
+
+    continuation = data.continuation
+  } while (continuation)
+}
+
+async function dumpObjAndBinaries(objData: ObjData) {
+  await dumpBinaries(objData)
+  dumpObj(objData)
+}
+
+async function dumpBinaries(data: ObjData | WidgetData) {
+  for (const value of Object.values(data))
+    if (isBinaryAttribute(value)) await dumpBinary(value[1].id)
+
+  const widgetPool = data._widget_pool || {}
+  for (const widget of Object.values(widgetPool)) await dumpBinaries(widget)
+}
+
+function isBinaryAttribute(data: unknown): data is ['binary', { id: string }] {
+  if (!Array.isArray(data)) return false
+  const [attributeType, attributeValue] = data
+  return (
+    attributeType === 'binary' &&
+    !!attributeValue &&
+    typeof attributeValue === 'object' &&
+    typeof attributeValue.id === 'string'
+  )
+}
+
+async function dumpBinary(binaryId: string) {
+  const binary = await fetchJson<BlobsData>(
+    `blobs/${encodeURIComponent(binaryId)}`,
+  )
+  const url = binary.private_access.get.url
+  const response = await fetch(url)
+  if (response.status !== 200) throw new Error(`Failed to fetch ${url}`)
+  const blob = await response.blob()
+  fs.writeFileSync(
+    `${BINARIES_PATH}/${encodeURIComponent(binaryId)}`,
+    Buffer.from(await blob.arrayBuffer()),
+  )
+}
+
+function dumpObj(objData: ObjData) {
+  fs.writeFileSync(
+    `${OBJS_PATH}/${objData._id}.json`,
+    JSON.stringify(objData, null, 2),
+  )
+}
+
+async function fetchJson<T>(
+  apiPath: string,
+  options: { data?: Record<string, unknown>; method?: string } = {},
+): Promise<T> {
+  process.stdout.write('.')
+
+  if (!apiToken) apiToken = await fetchIamToken()
+
+  const response = await fetch(
+    `https://api.scrivito.com/tenants/${INSTANCE_ID}/${apiPath}`,
+    {
+      body: options.data ? JSON.stringify(options.data) : undefined,
+      headers: {
+        Authorization: `Bearer ${apiToken}`,
+        'Content-Type': 'application/json',
+        'Scrivito-Access-As': 'editor',
```

Comment:
This looks strange. Why do we need to claim to be editor? Is this the "special header" that is missing in order for us to use `unstable_JrRestApi` directly?

Reply:
Yes, see the discussion:
https://infopark.slack.com/archives/G011EJEGKD5/p1704365149566769?thread_ts=1704362075.767339&cid=G011EJEGKD5

Also: https://infopark.slack.com/archives/C04TN8LJA/p1701704213052029?thread_ts=1701184869.334039&cid=C04TN8LJA

Reply:
I talked with Kris and `'Scrivito-Access-As': 'editor'` is more or less public API. I also filed a wish for `unstable_JrRestApi` to allow additional `headers`.